diff --git a/src/Bezosc.cpp b/src/Bezosc.cpp
index 8a08ef5..6737f6b 100644
--- a/src/Bezosc.cpp
+++ b/src/Bezosc.cpp
@@ -267,7 +267,7 @@ struct Bezosc : Module {
 struct BezoscWidget : ModuleWidget {
 	BezoscWidget(Bezosc* module) {
 		setModule(module);
-		setPanel(APP->window->loadSvg(asset::plugin(pluginInstance, "res/Bezosc.svg")));
+		setPanel(APP->window->loadSvg(asset::plugin(pluginInstance, "res/bezosc.svg")));
 
 		addChild(createWidget<ScrewSilver>(Vec(RACK_GRID_WIDTH, 0)));
 		addChild(createWidget<ScrewSilver>(Vec(box.size.x - 2 * RACK_GRID_WIDTH, 0)));
