diff --git a/Makefile b/Makefile
index 3a4d1d2..13791a6 100644
--- a/Makefile
+++ b/Makefile
@@ -2,9 +2,9 @@
 RACK_DIR ?= ../..
 
 # FLAGS will be passed to both the C and C++ compiler
-FLAGS += -Idep/include
+FLAGS +=
 CFLAGS +=
-CXXFLAGS +=
+CXXFLAGS += -Idep
 
 # Careful about linking to shared libraries, since you can't assume much about the user's environment and library search path.
 # Static libraries are fine, but they should be added to this plugin's build system.
diff --git a/src/javascript.hpp b/src/javascript.hpp
index 8425bab..98f01ef 100644
--- a/src/javascript.hpp
+++ b/src/javascript.hpp
@@ -34,7 +34,7 @@
 #include <rack.hpp>
 
 // QuickJS always throws a warning here, but it works.
-#include "quickjs/quickjs.h"
+#include "QuickJS/quickjs.h"
 
 namespace Javascript {
 
