diff --git a/Makefile b/Makefile
index 3a4d1d2..13791a6 100644
--- a/Makefile
+++ b/Makefile
@@ -43,6 +43,9 @@ else
 endif
 	QUICKJS_MAKE_FLAGS += CONFIG_DARWIN=y
 endif
+
+$(quickjs): DEP_CFLAGS := $(filter-out -march=armv7 -mfpu=neon -mno-unaligned-access,$(CFLAGS))
+
 $(quickjs):
 	cd dep && git clone "https://github.com/JerrySievert/QuickJS.git"
 	cd dep/QuickJS && git checkout b70d5344013836544631c361ae20569b978176c9
