diff --git a/Makefile b/Makefile
index 412bbfb..6fe0a98 100644
--- a/Makefile
+++ b/Makefile
@@ -23,6 +23,8 @@ DEPS += $(libsamplerate)
 $(libsamplerate):
 	$(WGET) http://www.mega-nerd.com/SRC/libsamplerate-0.1.9.tar.gz
 	cd dep && $(UNTAR) ../libsamplerate-0.1.9.tar.gz
+	cp /usr/share/automake-1.15/config.guess dep/libsamplerate-0.1.9/Cfg
+	cp /usr/share/automake-1.15/config.sub dep/libsamplerate-0.1.9/Cfg
 	cd dep/libsamplerate-0.1.9 && $(CONFIGURE)
 	cd dep/libsamplerate-0.1.9/src && $(MAKE)
 	cd dep/libsamplerate-0.1.9/src && $(MAKE) install
