diff --git a/include/panels/Panel.h b/include/panels/Panel.h
index 085a8227..89a2570f 100644
--- a/include/panels/Panel.h
+++ b/include/panels/Panel.h
@@ -60,11 +60,11 @@ public:
 
   static auto svg(const std::string &filename) -> std::shared_ptr<rack::Svg> {
     static const auto moduleAssetDir = pluginAssetDir() + P::moduleSlug + "/";
-    return rack::APP->window->loadSvg(moduleAssetDir + filename + ".svg");
+    return rack::appGet()->window->loadSvg(moduleAssetDir + filename + ".svg");
   }
 
   static auto panelSvg() -> std::shared_ptr<rack::Svg> {
-    return rack::APP->window->loadSvg(pluginAssetDir() + P::moduleSlug + ".svg");
+    return rack::appGet()->window->loadSvg(pluginAssetDir() + P::moduleSlug + ".svg");
   }
 
 protected:
