diff --git a/images/TrigDelay.png b/images/TrigDelay.png
index 31c1010..223de6c 100644
Binary files a/images/TrigDelay.png and b/images/TrigDelay.png differ
diff --git a/src/ML_components.cpp b/src/ML_components.cpp
index 372e926..eaf9e20 100644
--- a/src/ML_components.cpp
+++ b/src/ML_components.cpp
@@ -46,7 +46,7 @@ void MLSVGSwitch::onChange(const event::Change &e) {
 */
 
 
-WhiteLight::WhiteLight() {
+MLWhiteLight::MLWhiteLight() {
 	addBaseColor(color::WHITE);
 }
 
diff --git a/src/ML_components.hpp b/src/ML_components.hpp
index ecced7e..1c91625 100644
--- a/src/ML_components.hpp
+++ b/src/ML_components.hpp
@@ -47,8 +47,8 @@ struct MLSmallLight : BASE {
 	}
 };
 
-struct WhiteLight : ModuleLightWidget {
-	WhiteLight();
+struct MLWhiteLight : ModuleLightWidget {
+	MLWhiteLight();
 };
 
 
diff --git a/src/SH8.cpp b/src/SH8.cpp
index a0bc893..f5ae3b9 100644
--- a/src/SH8.cpp
+++ b/src/SH8.cpp
@@ -20,20 +20,20 @@ struct SH_channel {
 
 struct Random_Generator {
 
-	__m128i x1, x2;
+	simde__m128i x1, x2;
 	simd::float_4 fscale;
 
 	Random_Generator() {
 		fscale = float_4(10.0f/0xffffffffu);
-		x1 = _mm_set1_epi32(0x67452301);
-		x2 = _mm_setr_epi32(0xefcdab89, 0xefcdcb89, 0xefcdab29, 0xefcdab81);
+		x1 = simde_mm_set1_epi32(0x67452301);
+		x2 = simde_mm_setr_epi32(0xefcdab89, 0xefcdcb89, 0xefcdab29, 0xefcdab81);
 	};
 
 	~Random_Generator() {};
 
 	inline float_4 get() {
-		x1 = _mm_xor_si128(x1, x2);
-		x2 = _mm_add_epi32(x1, x2);
+		x1 = simde_mm_xor_si128(x1, x2);
+		x2 = simde_mm_add_epi32(x1, x2);
 		return simd::int32_4(x2) * fscale;
 	}
 };
diff --git a/src/simd_mask.hpp b/src/simd_mask.hpp
index 8dabbda..6655f50 100644
--- a/src/simd_mask.hpp
+++ b/src/simd_mask.hpp
@@ -9,11 +9,11 @@ struct ChannelMask {
 
     ChannelMask() {
         
-		__m128i tmp =  _mm_cmpeq_epi16(_mm_set_epi32(0,0,0,0),_mm_set_epi32(0,0,0,0));
+		simde__m128i tmp =  simde_mm_cmpeq_epi16(simde_mm_set_epi32(0,0,0,0),simde_mm_set_epi32(0,0,0,0));
 		
 		for(int i=0; i<4; i++) {
-			mask[3-i] = simd::float_4(_mm_castsi128_ps(tmp));
-			tmp = _mm_srli_si128(tmp, 4);
+			mask[3-i] = simd::float_4(simde_mm_castsi128_ps(tmp));
+			tmp = simde_mm_srli_si128(tmp, 4);
 		}
         
     };
