diff --git a/compile.mk b/compile.mk
index 81c55f3..017ee58 100644
--- a/compile.mk
+++ b/compile.mk
@@ -12,7 +12,7 @@ FLAGS += -MMD -MP
 # Debugger symbols. These are removed with `strip`.
 FLAGS += -g
 # Optimization
-FLAGS += -O3 -march=nocona -funsafe-math-optimizations
+FLAGS += -O3 -march=armv8-a -mtune=cortex-a53 -funsafe-math-optimizations
 # Warnings
 FLAGS += -Wall -Wextra -Wno-unused-parameter
 # C++ standard
@@ -72,7 +72,7 @@ build/%.m.o: %.m
 build/%.bin.o: %
 	@mkdir -p $(@D)
 ifdef ARCH_LIN
-	$(OBJCOPY) -I binary -O elf64-x86-64 -B i386:x86-64 --rename-section .data=.rodata,alloc,load,readonly,data,contents $< $@
+	$(OBJCOPY) -I binary -O elf64-littleaarch64 -B aarch64 --rename-section .data=.rodata,alloc,load,readonly,data,contents $< $@
 endif
 ifdef ARCH_WIN
 	$(OBJCOPY) -I binary -O pe-x86-64 -B i386:x86-64 --rename-section .data=.rodata,alloc,load,readonly,data,contents $< $@
diff --git a/dep.mk b/dep.mk
index bad90ec..6e14b73 100644
--- a/dep.mk
+++ b/dep.mk
@@ -5,7 +5,7 @@ DEP_LOCAL ?= dep
 $(shell mkdir -p $(DEP_LOCAL))
 DEP_PATH := $(abspath $(DEP_LOCAL))
 
-DEP_FLAGS += -g -O3 -march=nocona
+DEP_FLAGS += -g -O3 -march=armv8-a -mtune=cortex-a53
 
 ifeq ($(ARCH), mac)
 	DEP_MAC_SDK_FLAGS := -mmacosx-version-min=10.7
diff --git a/dep/Makefile b/dep/Makefile
index 35fbcd3..74145b9 100755
--- a/dep/Makefile
+++ b/dep/Makefile
@@ -170,6 +170,7 @@ speexdsp-SpeexDSP-1.2rc3:
 
 $(libspeexdsp): speexdsp-SpeexDSP-1.2rc3
 	cd speexdsp-SpeexDSP-1.2rc3 && $(CONFIGURE)
+	cp ../../resample_neon.h speexdsp-SpeexDSP-1.2rc3/libspeexdsp
 	$(MAKE) -C speexdsp-SpeexDSP-1.2rc3
 	$(MAKE) -C speexdsp-SpeexDSP-1.2rc3 install
 
@@ -180,6 +181,8 @@ libsamplerate-0.1.9:
 	rm libsamplerate-0.1.9.tar.gz
 
 $(libsamplerate): libsamplerate-0.1.9
+	cp /usr/share/automake-1.15/config.guess libsamplerate-0.1.9/Cfg
+	cp /usr/share/automake-1.15/config.sub libsamplerate-0.1.9/Cfg
 	cd libsamplerate-0.1.9 && $(CONFIGURE) --disable-fftw --disable-sndfile
 	$(MAKE) -C libsamplerate-0.1.9
 	$(MAKE) -C libsamplerate-0.1.9 install
diff --git a/src/engine/Engine.cpp b/src/engine/Engine.cpp
index ee0e64b..3b6bc33 100644
--- a/src/engine/Engine.cpp
+++ b/src/engine/Engine.cpp
@@ -10,7 +10,8 @@
 #include <mutex>
 #include <atomic>
 #include <tuple>
-#include <pmmintrin.h>
+/* #include <pmmintrin.h> */
+#include <unistd.h>
 
 
 namespace rack {
@@ -20,10 +21,10 @@ namespace engine {
 static void initMXCSR() {
 	// Set CPU to flush-to-zero (FTZ) and denormals-are-zero (DAZ) mode
 	// https://software.intel.com/en-us/node/682949
-	_MM_SET_FLUSH_ZERO_MODE(_MM_FLUSH_ZERO_ON);
-	_MM_SET_DENORMALS_ZERO_MODE(_MM_DENORMALS_ZERO_ON);
+	// _MM_SET_FLUSH_ZERO_MODE(_MM_FLUSH_ZERO_ON);
+	// _MM_SET_DENORMALS_ZERO_MODE(_MM_DENORMALS_ZERO_ON);
 	// Reset other flags
-	_MM_SET_ROUNDING_MODE(_MM_ROUND_NEAREST);
+	// _MM_SET_ROUNDING_MODE(_MM_ROUND_NEAREST);
 }
 
 
@@ -93,7 +94,7 @@ struct SpinBarrier {
 		}
 		else {
 			while (count != 0) {
-				_mm_pause();
+				asm volatile("yield" ::: "memory");
 			}
 		}
 	}
@@ -131,7 +132,7 @@ struct HybridBarrier {
 		while (!yield) {
 			if (count == 0)
 				return;
-			_mm_pause();
+			asm volatile("yield" ::: "memory");
 		}
 
 		// Wait on mutex
