diff --git a/compile.mk b/compile.mk
index 81c55f3..017ee58 100644
--- a/compile.mk
+++ b/compile.mk
@@ -12,7 +12,7 @@ FLAGS += -MMD -MP
 # Debugger symbols. These are removed with `strip`.
 FLAGS += -g
 # Optimization
-FLAGS += -O3 -march=nocona -funsafe-math-optimizations
+FLAGS += -O3 -march=native -mcpu=native -funsafe-math-optimizations
 # Warnings
 FLAGS += -Wall -Wextra -Wno-unused-parameter
 # C++ standard
@@ -72,7 +72,7 @@ build/%.m.o: %.m
 build/%.bin.o: %
 	@mkdir -p $(@D)
 ifdef ARCH_LIN
-	$(OBJCOPY) -I binary -O elf64-x86-64 -B i386:x86-64 --rename-section .data=.rodata,alloc,load,readonly,data,contents $< $@
+	$(OBJCOPY) -I binary -O elf64-littleaarch64 -B aarch64 --rename-section .data=.rodata,alloc,load,readonly,data,contents $< $@
 endif
 ifdef ARCH_WIN
 	$(OBJCOPY) -I binary -O pe-x86-64 -B i386:x86-64 --rename-section .data=.rodata,alloc,load,readonly,data,contents $< $@
diff --git a/dep.mk b/dep.mk
index bad90ec..6e14b73 100644
--- a/dep.mk
+++ b/dep.mk
@@ -5,7 +5,7 @@ DEP_LOCAL ?= dep
 $(shell mkdir -p $(DEP_LOCAL))
 DEP_PATH := $(abspath $(DEP_LOCAL))
 
-DEP_FLAGS += -g -O3 -march=nocona
+DEP_FLAGS += -g -O3 -march=native -mcpu=native
 
 ifeq ($(ARCH), mac)
 	DEP_MAC_SDK_FLAGS := -mmacosx-version-min=10.7
diff --git a/dep/Makefile b/dep/Makefile
index 33d45e6..c07b301 100755
--- a/dep/Makefile
+++ b/dep/Makefile
@@ -92,6 +92,7 @@ $(libspeexdsp):
 	$(SHA256) speexdsp-SpeexDSP-1.2rc3.tgz c8dded1454747f65956f981c95e7f89a06abdaa2a53e8aeaa66bab2a3d59cebd
 	$(UNTAR) speexdsp-SpeexDSP-1.2rc3.tgz
 	cd speexdsp-SpeexDSP-1.2rc3 && $(CONFIGURE)
+	cp ../../resample_neon.h speexdsp-SpeexDSP-1.2rc3/libspeexdsp
 	$(MAKE) -C speexdsp-SpeexDSP-1.2rc3
 	$(MAKE) -C speexdsp-SpeexDSP-1.2rc3 install
 
diff --git a/src/engine/Engine.cpp b/src/engine/Engine.cpp
index a16d5a0..2e34a09 100644
--- a/src/engine/Engine.cpp
+++ b/src/engine/Engine.cpp
@@ -10,7 +10,8 @@
 #include <mutex>
 #include <atomic>
 #include <tuple>
-#include <x86intrin.h>
+/* #include <x86intrin.h> */
+#include <unistd.h>
 
 
 namespace rack {
@@ -20,8 +22,8 @@ namespace engine {
 static void disableDenormals() {
 	// Set CPU to flush-to-zero (FTZ) and denormals-are-zero (DAZ) mode
 	// https://software.intel.com/en-us/node/682949
-	_MM_SET_FLUSH_ZERO_MODE(_MM_FLUSH_ZERO_ON);
-	_MM_SET_DENORMALS_ZERO_MODE(_MM_DENORMALS_ZERO_ON);
+	// _MM_SET_FLUSH_ZERO_MODE(_MM_FLUSH_ZERO_ON);
+	// _MM_SET_DENORMALS_ZERO_MODE(_MM_DENORMALS_ZERO_ON);
 }
 
 
@@ -91,7 +93,7 @@ struct SpinBarrier {
 		}
 		else {
 			while (count != 0) {
-				_mm_pause();
+				asm volatile("yield" ::: "memory");
 			}
 		}
 	}
@@ -129,7 +131,7 @@ struct HybridBarrier {
 		while (!yield) {
 			if (count == 0)
 				return;
-			_mm_pause();
+			asm volatile("yield" ::: "memory");
 		}
 
 		// Wait on mutex
diff --git a/src/window.cpp b/src/window.cpp
index 02082ec..ce5c502 100644
--- a/src/window.cpp
+++ b/src/window.cpp
@@ -245,6 +245,10 @@ Window::Window() {
 	glfwSetKeyCallback(win, keyCallback);
 	glfwSetDropCallback(win, dropCallback);
 
+#if (defined(__arm__) || defined(__aarch64__))
+	glfwIconifyWindow(win);
+#endif
+
 	// Set up GLEW
 	glewExperimental = GL_TRUE;
 	err = glewInit();
