diff --git a/Makefile b/Makefile
index 7c4d4e5..d587bdf 100644
--- a/Makefile
+++ b/Makefile
@@ -66,9 +66,7 @@ ifdef ARCH_WIN
 	FLAGS += -fmax-errors=5
 endif
 
-#  -flto
-FLAGS += -finline-limit=500000 -finline-functions-called-once -flto
-LDFLAGS += -flto
+FLAGS += -finline-limit=500000 -finline-functions-called-once
 
 include test.mk
 
diff --git a/test.mk b/test.mk
index c34c0b2..35159d6 100644
--- a/test.mk
+++ b/test.mk
@@ -29,7 +29,6 @@ perf.exe : PERFFLAG = -D _PERF
 perf.exe : FLAGS += $(ASSERTOFF)
 
 FLAGS += $(PERFFLAG)
-FLAGS += -flto
 
 test.exe : FLAGS += -D _TESTEX
 
