diff --git a/src/atsr.cpp b/src/atsr.cpp
index 5287931..2696555 100644
--- a/src/atsr.cpp
+++ b/src/atsr.cpp
@@ -512,10 +512,10 @@ struct AtsrWidget : ModuleWidget  {
         addOutput(createOutput<ViaJack>(Vec(83.68 + 1.053, 324.02), module, Atsr::MAIN_OUTPUT));
         addOutput(createOutput<ViaJack>(Vec(135 + 1.053, 241.12), module, Atsr::AUX_LOGIC_OUTPUT));
 
-        addChild(createLight<MediumLight<WhiteLight>>(Vec(35.8 + .753, 268.5), module, Atsr::LED1_LIGHT));
-        addChild(createLight<MediumLight<WhiteLight>>(Vec(73.7 + .753, 268.5), module, Atsr::LED2_LIGHT));
-        addChild(createLight<MediumLight<WhiteLight>>(Vec(35.8 + .753, 309.9), module, Atsr::LED3_LIGHT));
-        addChild(createLight<MediumLight<WhiteLight>>(Vec(73.7 + .753, 309.9), module, Atsr::LED4_LIGHT));
+        addChild(createLight<MediumLight<ViaWhiteLight>>(Vec(35.8 + .753, 268.5), module, Atsr::LED1_LIGHT));
+        addChild(createLight<MediumLight<ViaWhiteLight>>(Vec(73.7 + .753, 268.5), module, Atsr::LED2_LIGHT));
+        addChild(createLight<MediumLight<ViaWhiteLight>>(Vec(35.8 + .753, 309.9), module, Atsr::LED3_LIGHT));
+        addChild(createLight<MediumLight<ViaWhiteLight>>(Vec(73.7 + .753, 309.9), module, Atsr::LED4_LIGHT));
         addChild(createLight<MediumLight<GreenRedLight>>(Vec(54.8 + .753, 179.6), module, Atsr::OUTPUT_GREEN_LIGHT));
         addChild(createLight<LargeLight<RGBTriangle>>(Vec(59 + .753, 221), module, Atsr::RED_LIGHT));
 
diff --git a/src/gateseq.cpp b/src/gateseq.cpp
index d2460b3..cb14ffa 100644
--- a/src/gateseq.cpp
+++ b/src/gateseq.cpp
@@ -568,10 +568,10 @@ struct GateseqWidget : ModuleWidget  {
         addOutput(createOutput<ViaJack>(Vec(83.68 + 1.053, 324.02), module, Gateseq::MAIN_OUTPUT));
         addOutput(createOutput<ViaJack>(Vec(135 + 1.053, 241.12), module, Gateseq::AUX_LOGIC_OUTPUT));
 
-        addChild(createLight<MediumLight<WhiteLight>>(Vec(35.8 + .753, 268.5), module, Gateseq::LED1_LIGHT));
-        addChild(createLight<MediumLight<WhiteLight>>(Vec(73.7 + .753, 268.5), module, Gateseq::LED2_LIGHT));
-        addChild(createLight<MediumLight<WhiteLight>>(Vec(35.8 + .753, 309.9), module, Gateseq::LED3_LIGHT));
-        addChild(createLight<MediumLight<WhiteLight>>(Vec(73.7 + .753, 309.9), module, Gateseq::LED4_LIGHT));
+        addChild(createLight<MediumLight<ViaWhiteLight>>(Vec(35.8 + .753, 268.5), module, Gateseq::LED1_LIGHT));
+        addChild(createLight<MediumLight<ViaWhiteLight>>(Vec(73.7 + .753, 268.5), module, Gateseq::LED2_LIGHT));
+        addChild(createLight<MediumLight<ViaWhiteLight>>(Vec(35.8 + .753, 309.9), module, Gateseq::LED3_LIGHT));
+        addChild(createLight<MediumLight<ViaWhiteLight>>(Vec(73.7 + .753, 309.9), module, Gateseq::LED4_LIGHT));
         addChild(createLight<MediumLight<GreenRedLight>>(Vec(54.8 + .753, 179.6), module, Gateseq::OUTPUT_GREEN_LIGHT));
         addChild(createLight<LargeLight<RGBTriangle>>(Vec(59 + .753, 221), module, Gateseq::RED_LIGHT));
 
diff --git a/src/meta.cpp b/src/meta.cpp
index 8fc4d68..7c9c0cc 100644
--- a/src/meta.cpp
+++ b/src/meta.cpp
@@ -821,10 +821,10 @@ struct MetaWidget : ModuleWidget  {
         addOutput(createOutput<ViaJack>(Vec(83.68 + 1.053, 324.02), module, Meta::MAIN_OUTPUT));
         addOutput(createOutput<ViaJack>(Vec(135 + 1.053, 241.12), module, Meta::AUX_LOGIC_OUTPUT));
 
-        addChild(createLight<MediumLight<WhiteLight>>(Vec(35.8 + .753, 268.5), module, Meta::LED1_LIGHT));
-        addChild(createLight<MediumLight<WhiteLight>>(Vec(73.1 + .753, 268.5), module, Meta::LED2_LIGHT));
-        addChild(createLight<MediumLight<WhiteLight>>(Vec(35.8 + .753, 309.9), module, Meta::LED3_LIGHT));
-        addChild(createLight<MediumLight<WhiteLight>>(Vec(73.1 + .753, 309.9), module, Meta::LED4_LIGHT));
+        addChild(createLight<MediumLight<ViaWhiteLight>>(Vec(35.8 + .753, 268.5), module, Meta::LED1_LIGHT));
+        addChild(createLight<MediumLight<ViaWhiteLight>>(Vec(73.1 + .753, 268.5), module, Meta::LED2_LIGHT));
+        addChild(createLight<MediumLight<ViaWhiteLight>>(Vec(35.8 + .753, 309.9), module, Meta::LED3_LIGHT));
+        addChild(createLight<MediumLight<ViaWhiteLight>>(Vec(73.1 + .753, 309.9), module, Meta::LED4_LIGHT));
         addChild(createLight<MediumLight<GreenRedLight>>(Vec(54.8 + .753, 179.6), module, Meta::OUTPUT_GREEN_LIGHT));
         addChild(createLight<LargeLight<RGBTriangle>>(Vec(59 + .753, 221), module, Meta::RED_LIGHT));
 
diff --git a/src/osc3.cpp b/src/osc3.cpp
index 5952a99..3f349d6 100644
--- a/src/osc3.cpp
+++ b/src/osc3.cpp
@@ -203,10 +203,10 @@ struct Osc3Widget : ModuleWidget  {
         addOutput(createOutput<ViaJack>(Vec(83.68 + 1.053, 324.02), module, Osc3::MAIN_OUTPUT));
         addOutput(createOutput<ViaJack>(Vec(135 + 1.053, 241.12), module, Osc3::AUX_LOGIC_OUTPUT));
 
-        addChild(createLight<MediumLight<WhiteLight>>(Vec(35.8 + .753, 268.5), module, Osc3::LED1_LIGHT));
-        addChild(createLight<MediumLight<WhiteLight>>(Vec(73.1 + .753, 268.5), module, Osc3::LED2_LIGHT));
-        addChild(createLight<MediumLight<WhiteLight>>(Vec(35.8 + .753, 309.9), module, Osc3::LED3_LIGHT));
-        addChild(createLight<MediumLight<WhiteLight>>(Vec(73.1 + .753, 309.9), module, Osc3::LED4_LIGHT));
+        addChild(createLight<MediumLight<ViaWhiteLight>>(Vec(35.8 + .753, 268.5), module, Osc3::LED1_LIGHT));
+        addChild(createLight<MediumLight<ViaWhiteLight>>(Vec(73.1 + .753, 268.5), module, Osc3::LED2_LIGHT));
+        addChild(createLight<MediumLight<ViaWhiteLight>>(Vec(35.8 + .753, 309.9), module, Osc3::LED3_LIGHT));
+        addChild(createLight<MediumLight<ViaWhiteLight>>(Vec(73.1 + .753, 309.9), module, Osc3::LED4_LIGHT));
         addChild(createLight<MediumLight<GreenRedLight>>(Vec(54.8 + .753, 179.6), module, Osc3::OUTPUT_GREEN_LIGHT));
         addChild(createLight<LargeLight<RGBTriangle>>(Vec(59 + .753, 221), module, Osc3::RED_LIGHT));
 
diff --git a/src/scanner.cpp b/src/scanner.cpp
index 86f82d6..b2e1f78 100644
--- a/src/scanner.cpp
+++ b/src/scanner.cpp
@@ -319,10 +319,10 @@ struct ScannerWidget : ModuleWidget  {
         addOutput(createOutput<ViaJack>(Vec(83.68 + 1.053, 324.02), module, Scanner::MAIN_OUTPUT));
         addOutput(createOutput<ViaJack>(Vec(135 + 1.053, 241.12), module, Scanner::AUX_LOGIC_OUTPUT));
 
-        addChild(createLight<MediumLight<WhiteLight>>(Vec(35.8 + .753, 268.5), module, Scanner::LED1_LIGHT));
-        addChild(createLight<MediumLight<WhiteLight>>(Vec(73.7 + .753, 268.5), module, Scanner::LED2_LIGHT));
-        addChild(createLight<MediumLight<WhiteLight>>(Vec(35.8 + .753, 309.9), module, Scanner::LED3_LIGHT));
-        addChild(createLight<MediumLight<WhiteLight>>(Vec(73.7 + .753, 309.9), module, Scanner::LED4_LIGHT));
+        addChild(createLight<MediumLight<ViaWhiteLight>>(Vec(35.8 + .753, 268.5), module, Scanner::LED1_LIGHT));
+        addChild(createLight<MediumLight<ViaWhiteLight>>(Vec(73.7 + .753, 268.5), module, Scanner::LED2_LIGHT));
+        addChild(createLight<MediumLight<ViaWhiteLight>>(Vec(35.8 + .753, 309.9), module, Scanner::LED3_LIGHT));
+        addChild(createLight<MediumLight<ViaWhiteLight>>(Vec(73.7 + .753, 309.9), module, Scanner::LED4_LIGHT));
         addChild(createLight<MediumLight<GreenRedLight>>(Vec(54.8 + .753, 179.6), module, Scanner::OUTPUT_GREEN_LIGHT));
         addChild(createLight<LargeLight<RGBTriangle>>(Vec(59 + .753, 221), module, Scanner::RED_LIGHT));
 
diff --git a/src/sync.cpp b/src/sync.cpp
index ff4fab2..084bc93 100644
--- a/src/sync.cpp
+++ b/src/sync.cpp
@@ -534,10 +534,10 @@ struct Sync_Widget : ModuleWidget  {
         addOutput(createOutput<ViaJack>(Vec(83.68 + 1.053, 324.02), module, Sync::MAIN_OUTPUT));
         addOutput(createOutput<ViaJack>(Vec(135 + 1.053, 241.12), module, Sync::AUX_LOGIC_OUTPUT));
 
-        addChild(createLight<MediumLight<WhiteLight>>(Vec(35.8 + .753, 268.5), module, Sync::LED1_LIGHT));
-        addChild(createLight<MediumLight<WhiteLight>>(Vec(73.7 + .753, 268.5), module, Sync::LED2_LIGHT));
-        addChild(createLight<MediumLight<WhiteLight>>(Vec(35.8 + .753, 309.9), module, Sync::LED3_LIGHT));
-        addChild(createLight<MediumLight<WhiteLight>>(Vec(73.7 + .753, 309.9), module, Sync::LED4_LIGHT));
+        addChild(createLight<MediumLight<ViaWhiteLight>>(Vec(35.8 + .753, 268.5), module, Sync::LED1_LIGHT));
+        addChild(createLight<MediumLight<ViaWhiteLight>>(Vec(73.7 + .753, 268.5), module, Sync::LED2_LIGHT));
+        addChild(createLight<MediumLight<ViaWhiteLight>>(Vec(35.8 + .753, 309.9), module, Sync::LED3_LIGHT));
+        addChild(createLight<MediumLight<ViaWhiteLight>>(Vec(73.7 + .753, 309.9), module, Sync::LED4_LIGHT));
         addChild(createLight<MediumLight<GreenRedLight>>(Vec(54.8 + .753, 179.6), module, Sync::OUTPUT_GREEN_LIGHT));
         addChild(createLight<LargeLight<RGBTriangle>>(Vec(59 + .753, 221), module, Sync::RED_LIGHT));
 
diff --git a/src/via_ui.hpp b/src/via_ui.hpp
index b187c2d..00d9b5a 100644
--- a/src/via_ui.hpp
+++ b/src/via_ui.hpp
@@ -6,8 +6,8 @@
 
 // Modified light widget for the white LED
 
-struct WhiteLight : ModuleLightWidget {
-    WhiteLight() {
+struct ViaWhiteLight : ModuleLightWidget {
+    ViaWhiteLight() {
         addBaseColor(SCHEME_WHITE);
     }
 };
