diff --git a/src/modules/Iverson.cpp b/src/modules/Iverson.cpp
index bf3919d..3731afa 100644
--- a/src/modules/Iverson.cpp
+++ b/src/modules/Iverson.cpp
@@ -652,8 +652,8 @@ namespace sspo
         onSampleRateChange();
         iverson->init();
 
-        controllerPageUpdateDivider.setDivision (2048);
-        paramMidiUpdateDivider.setDivision (128);
+        controllerPageUpdateDivider.setDivision (1024);
+        paramMidiUpdateDivider.setDivision (64);
     }
 
     struct Iverson : IversonBase
