diff --git a/Makefile b/Makefile
index 893666b..4378ee9 100644
--- a/Makefile
+++ b/Makefile
@@ -20,7 +20,7 @@ endif
 
 # Careful about linking to shared libraries, since you can't assume much about the user's environment and library search path.
 # Static libraries are fine, but they should be added to this plugin's build system.
-LDFLAGS += -lpthread
+LDFLAGS += -lpthread /usr/lib/gcc/arm-linux-gnueabihf/8/libatomic.a
 
 # Add .cpp files to the build
 SOURCES += $(wildcard src/modules/*.cpp)
