diff --git a/src/common/vt_dsp/lipol.cpp b/src/common/vt_dsp/lipol.cpp
index ef4744fc..65b50f80 100644
--- a/src/common/vt_dsp/lipol.cpp
+++ b/src/common/vt_dsp/lipol.cpp
@@ -15,11 +15,11 @@ lipol_ps::lipol_ps()
 
 void lipol_ps::set_blocksize(int bs)
 {
-#ifdef ARM_NEON
-   lipol_BLOCK_SIZE = vdupq_n_f32(bs);
-#else
+// #ifdef ARM_NEON
+//    lipol_BLOCK_SIZE = vdupq_n_f32(bs);
+// #else
    lipol_BLOCK_SIZE = _mm_cvt_si2ss(m128_zero, bs);
-#endif
+// #endif
    m128_bs4_inv = _mm_div_ss(m128_four, lipol_BLOCK_SIZE);
 }
 
