diff --git a/Makefile b/Makefile
index 8b52374..4414310 100644
--- a/Makefile
+++ b/Makefile
@@ -99,14 +99,14 @@ LDFLAGS += -lwinmm
 endif
 
 ifdef ARCH_LIN
-SOURCES += surge/src/linux/ConfigurationXml.S
+SOURCES += surge/src/linux/ConfigurationXmlARM.S
 LDFLAGS += -lstdc++fs -pthread
 FLAGS += -DUSE_STD_EXPERIMENTAL_FILESYSTEM
 
 # This is really a hack but...
-build/surge/src/linux/ConfigurationXml.S.o: surge/src/linux/ConfigurationXml.S
+build/surge/src/linux/ConfigurationXmlARM.S.o: surge/src/linux/ConfigurationXmlARM.S
 	mkdir -p build/surge/src/linux
-	cd surge/src/linux && $(CC) -c ConfigurationXml.S -o ../../../$@
+	cd surge/src/linux && $(CC) -c ConfigurationXmlARM.S -o ../../../$@
 
 endif
 
@@ -144,7 +144,7 @@ ifdef ARCH_MAC
 endif
 
 ifdef ARCH_LIN
-	FLAGS += -DLINUX -D"_aligned_malloc(x,a)=malloc(x)" -D"_aligned_free(x)=free(x)"
+	FLAGS += -DLINUX -D"_aligned_malloc(x,a)=aligned_alloc(a,x)" -D"_aligned_free(x)=free(x)"
 	FLAGS += -Wno-nonnull-compare \
 	-Wno-sign-compare \
 	-Wno-char-subscripts \
@@ -154,6 +154,14 @@ ifdef ARCH_LIN
 	-Wno-multichar
 
 	FLAGS += -Isurge/src/linux
+
+	FLAGS += -DARM_NEON \
+	-DSIMDE_ENABLE_NATIVE_ALIASES \
+	-march=armv8-a \
+	-mtune=cortex-a53 \
+	-Wno-psabi \
+	-flax-vector-conversions \
+	-Isurge/libs/simde
 endif
 
 ifdef ARCH_WIN
