diff --git a/Makefile b/Makefile
index a7a99ac..b47abba 100644
--- a/Makefile
+++ b/Makefile
@@ -12,7 +12,7 @@ VERSION = 1.0
 
 
 # FLAGS will be passed to both the C and C++ compiler
-FLAGS += -std=c++17 #-faligned-allocation -faligned-new# -fopt-info-loop-optimized # -fopt-info-vec-missed
+FLAGS += #-std=c++17 -faligned-allocation -faligned-new# -fopt-info-loop-optimized # -fopt-info-vec-missed
 CFLAGS +=
 CXXFLAGS += -Isrc -Isrc/dsp -Isrc/models
 
